src/azul/drs.py

```diff
@@ -74,10 +87,133 @@ class Access:
     headers: Optional[Mapping[str, str]] = None
 
 
+class DRSURI(metaclass=ABCMeta):
+    scheme: str
+    netloc: str
+    object_id: str
```

Comment:
Only expose the properties that are really needed at the usage sites. I am not sure all three are needed at the usage sites. Also consider `@abstractproperty` for those that are.